### PR TITLE
PD-2168, PD-2149: Refresh Menu

### PR DESCRIPTION
**Pull request by shaneeza**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [PD-2168](https://jira.mongodb.org/browse/PD-2168), [PD-2194](https://jira.mongodb.org/browse/PD-2194)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Style change (purely visual updates; if this is the case, attach a screenshot below!)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

## Feedback Levels

Please try to mark your comments with one of these levels

1. Praise (👏 🚀 🔥 🍾)
2. Opinions, Ideas, Suggestions (💡 👀)
3. Let’s discuss @ Code review (🤷 🤔 🧐 )
4. Doesn’t meet agreed upon standards (😬 ⚠️ ‼️)
5. Missing an edge case (🛑 🙅 ❌)
6. Missing core functionality (⛔️ ☣️ 🚧 🌋 💀)

(Note: Use ⌘-⌃-␣ to open the Emoji keyboard!)

### CODE REVIEW

**Comment by TheSonOfThomp** [comment]
Are all sub-items supposed to show the active wedge? This is in Light-mode as well.

Screen Shot 2022-08-02 at 12 02 59 PM

File: packages/menu/src/Menu.tsx [hunk 1]
```diff
@@ -98,7 +112,7 @@ function Menu({
 
     function updateChildren(children: React.ReactNode): React.ReactNode {
       return React.Children.map(children, child => {
-        if (!React.isValidElement(child) || child.props?.disabled) {
+        if (!React.isValidElement(child)) {
```
Comment (TheSonOfThomp) [inline packages/menu/src/Menu.tsx:115]: Should we highlight this change as well?

**Comment by shaneeza** [comment]
@TheSonOfThomp i think you might be right, now that i think about it a submenu item would never be active? i'll double check with @audroue

File: packages/menu/src/MenuItem.tsx [hunk 2]
```diff
@@ -63,6 +65,13 @@ interface BaseMenuItemProps {
   children?: React.ReactNode;
 
   href?: string;
+
+  /**
+   * Determines whether or not the component will be rendered in dark mode.
+   *
+   * default: `false`
+   */
+  darkMode?: boolean;
```
Comment (TheSonOfThomp) [inline packages/menu/src/MenuItem.tsx:74]: I think we should use an internal `MenuContext` for this, or use an `InternalMenuItem` component like we do with `Select`. It feels confusing to have a consumer-facing prop like this that doesn't do anything

File: packages/menu/src/SubMenu.tsx [hunk 3]
```diff
@@ -190,6 +315,18 @@ interface SubMenuProps {
   onExited?: ExitHandler<HTMLElement>;
 
   href?: string;
+
+  /**
+   * Determines whether or not the component will be rendered in dark Theme.
+   *
+   * default: `false`
+   */
+  darkMode?: boolean;
```
Comment (TheSonOfThomp) [inline packages/menu/src/SubMenu.tsx:324]: Same concern here. Unless the consumer can toggle DarkMode for individual SubMenus/MenuItems, I don't think we should have this as a prop

File: packages/menu/src/styles.ts [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
 export const svgWidth = 24;
-export const menuItemPadding = 15;
-export const paddingLeft = 52;
-const indentation = 16;
+export const paddingLeftWithGlyph = 54;
+export const paddingLeftWithoutGlyph = 20;
+const indentation = 20;
 const wedgeWidth = 4;
 const minMenuContent = 32;
 const menuBlockPadding: Record<Size, number> = {
```
Comment (TheSonOfThomp) [inline packages/menu/src/styles.ts:13]: These seem to be unused

**Review by TheSonOfThomp — CHANGES_REQUESTED** [review]
Styles look good. I have a concern about the external API though

File: packages/menu/src/MenuItem.tsx (hunk 2 above)
Comment (shaneeza) [reply packages/menu/src/MenuItem.tsx:74]: I see, thats similar to what the select component. Yeah that makes more sense.

File: packages/menu/src/Menu.tsx (hunk 1 above)
Comment (shaneeza) [reply packages/menu/src/Menu.tsx:115]: i'll add it

File: packages/menu/src/MenuItem.tsx (hunk 2 above)
Comment (spark33) [reply packages/menu/src/MenuItem.tsx:74]: We also have an extendable `DarkModeProps` interface now! For all places in this PR that define `darkMode` in interfaces

File: packages/menu/src/MenuItem.tsx [hunk 5]
```diff
@@ -86,6 +96,9 @@ const MenuItem: ExtendableBox<
     const hoverStyles = getHoverStyles(menuItemContainer.selector);
     const focusStyles = getFocusedStyles(menuItemContainer.selector);
 
+    // Darkmode is passed from the `Menu` component which is using the `useDarkMode` hook
+    const theme = darkMode ? Theme.Dark : Theme.Light;
```
Comment (spark33) [inline packages/menu/src/MenuItem.tsx:100]: If we move to using a `MenuContext`, the `useDarkMode` hook also supports:

`const { darkMode, theme } = useDarkMode(darkModeProp);` so all children components could just consume this from `MenuContext`

**Review by spark33 — CHANGES_REQUESTED** [review]
Minor comments around the usage of the recent darkMode utilities

**Comment by shaneeza** [comment]
**note to self: update website**

File: packages/menu/src/Menu.tsx (hunk 1 above)
Comment (shaneeza) [reply packages/menu/src/Menu.tsx:116]: Actually i'm going to add this back. If it's disabled it doesn't need to be cloned or have any additional props added to it.

**Comment by shaneeza** [comment]
> Are all sub-items supposed to show the active wedge? This is in Light-mode as well.

@TheSonOfThomp i checked with Rob, submenu items can be active and should show the little wedge. I will clean up the example a bit so there is are not three active state.

**Review by TheSonOfThomp — APPROVED** [review]
LGTM! 🎸

**Review by spark33 — APPROVED** [review]
noiiice

File: packages/menu/src/MenuContext.tsx [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import { createContext } from 'react';
+import { Theme } from '@leafygreen-ui/lib';
+
+interface MenuData {
+  theme: Theme;
+}
+
+export const MenuContext = createContext<MenuData>({
+  theme: Theme.Light,
+});
```
Comment (TheSonOfThomp) [inline packages/menu/src/MenuContext.tsx:10]: There are probably other props we're passing in that could just be in Context, but that doesn't need to be done right now

Comment (shaneeza) [reply packages/menu/src/MenuContext.tsx:10]: Yeah i was thinking about the `size` prop while I was adding this.

**Pull request merged**